spot-contracts/contracts/RolloverVault.sol

```diff
@@ -709,6 +751,46 @@ contract RolloverVault is
     //--------------------------------------------------------------------------
     // Private write methods
 
+    /// @dev Executes a system-level rebalance. This operation transfers value between
+    ///      the perp reserve and the vault such that the system moves toward balance.
+    ///      Settles protocol fees, charged as a percentage of the bi-directional flow in value.
+    ///      Performs some book-keeping to keep track of the vault and perp's assets.
+    function _rebalance(IPerpetualTranche perp_, IERC20Upgradeable underlying_) private {
+        // If the vault has any perp tokens we deconstruct it
+        // such that it's value reflects in the vault's TVL,
+        // before the rebalance calculation.
+        _meldPerps(perp_);
+
+        SubscriptionParams memory s = _querySubscriptionState(perp_);
+        RebalanceData memory r = feePolicy.computeRebalanceData(s);
+        if (r.underlyingAmtIntoPerp <= 0) {
+            // We transfer value from perp to the vault, by minting the vault perp tokens.
+            // NOTE: We first mint the vault perp tokens, and then pay the protocol fee.
+            uint256 perpSupply = perp_.totalSupply();
+            uint256 perpAmtToVault = r.underlyingAmtIntoPerp.abs().mulDiv(perpSupply, s.perpTVL);
```

Comment:
I still think it's better design to structure it like:
perp.debase(underlyingAmt)
or even better naming-- rebalanceToVault(underlyingAmt).

Since we're starting with underlying amounts here anyway.

Let perp handle the perp math and how the value transfer is implemented. Let the vault not have to worry about it.

Reply:
Yes that's simpler